### DISCUSSION TITLE
ToArray(Async) vs ToList(Async) in Entity Framework 8

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#120] [General]
# ToArray(Async) vs ToList(Async) in Entity Framework 8

When retrieving data from your database with Entity Framework, there are two major options: ToArray and ToList. Besides the different return type, is there any significant difference in performance between the two? Let's find out!

https://steven-giesel.com/blogPost/4ec751cd-d773-44c2-996e-cc4ae9ae966c/toarrayasync-vs-tolistasync-in-entity-framework-8?utm_source=devdigest.today&utm_medium=website&utm_campaign=feature_promo&utm_content=link_click

**Comment by aunikitin** [comment]
Probably, it worth watching what is going underhood and life become match more simple:

public static async Task<TSource[]> ToArrayAsync<TSource>(
        this IQueryable<TSource> source,
        CancellationToken cancellationToken = default)
        => (await source.ToListAsync(cancellationToken).ConfigureAwait(false)).ToArray();

**Reply by linkdotnet** [reply]
Thanks for the hint! I added your comment in the post! Much appreciated

**Comment by danielmarbach** [comment]
I assume the actual benchmark used Number Of Elements by passing it to the seeder? 😊

**Reply by linkdotnet** [reply]
Yeah exactly.
